### PR TITLE
Add class "external-link" to external links

### LINKED ISSUE #881: Distinguish local and remote links with a class
And then style so that you can tell when you're going to a remote link.

### PR DESCRIPTION
**Pull request by maelle**
Fix #881

### CODE REVIEW

**Comment by maelle** [comment]
I think further refinement of classes needs to happen in downlit?

File: R/html-tweak.R [hunk 1]
```diff
@@ -71,6 +71,32 @@ tweak_md_links <- function(html) {
...
 }
 
+tweak_all_links <- function(html, pkg = pkg) {
+  links <- xml2::xml_find_all(html, ".//a")
+  if (length(links) == 0)
+    return()
+
+  hrefs <- xml2::xml_attr(links, "href")
+
+  if (is.null(pkg$meta$url)) {
```
Comment (hadley) [inline R/html-tweak.R:81]: Do we actually use this to create absolute urls?

Comment (maelle) [reply R/html-tweak.R:81]: No but in e.g. the README there are absolute URLs (to the homepage, to the code of conduct) https://github.com/r-lib/pkgdown#pkgdown- so one needs to know the URL to not detect those as external links

Comment (hadley) [reply R/html-tweak.R:81]: Ah good point. A comment mentioning that would be useful for future-me 😄

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # pkgdown (development version)
 
+* External links now get the class "external-link" on top of their usual classes, if they had any. 
+Therefore they can be targeted more easily by CSS rules. (#881, #1491)
```
Comment (hadley) [inline NEWS.md:4]: Needs indent here

File: R/html-tweak.R (hunk 1 above)
Comment (hadley) [reply R/html-tweak.R:82]: The comment currently tells me what the code does — I want to know _why_ we need to do it (i.e. READMEs often include absolute urls for code of conduct etc)

File: R/html-tweak.R [hunk 3]
```diff
@@ -71,6 +71,33 @@ tweak_md_links <- function(html) {
...
+  # Use pkgdown URL to detect non-external absolute URLs
+  if (is.null(pkg$meta$url)) {
+    needs_tweak <- grepl("https?\\:\\/\\/", hrefs)
+  } else {
+    url <- sub("/$", "", pkg$meta$url)
+    needs_tweak <- grepl("https?\\:\\/\\/", hrefs) & !grepl(url, hrefs)
+  }
+
+  if (any(needs_tweak)) {
+    tweaked <- purrr::map(
```
Comment (hadley) [inline R/html-tweak.R:90]: Can this fit on one line?

File: R/rmarkdown.R [hunk 4]
```diff
@@ -45,7 +45,12 @@ render_rmarkdown <- function(pkg, input, output, ..., copy_images = TRUE, quiet
   )
 
   if (identical(path_ext(path)[[1]], "html")) {
-    update_html(path, tweak_rmarkdown_html, input_path = path_dir(input_path))
+    update_html(
+      path,
+      tweak_rmarkdown_html,
+      input_path = path_dir(input_path),
+      pkg = pkg
+      )
```
Comment (hadley) [inline R/rmarkdown.R:53]: ◀️

File: tests/testthat/test-markdown.R [hunk 5]
```diff
@@ -1,11 +1,16 @@
 test_that("empty string works", {
   skip_if_not(rmarkdown::pandoc_available())
-  expect_equal(markdown_text(""), "")
+  expect_equal(
+    markdown_text("", pkg = list(meta = list(url = "https://example.com"))),
+    "")
```
Comment (hadley) [inline tests/testthat/test-markdown.R:5]: `)` should go on own line so it can match indent of opening `(`

**Review by hadley — APPROVED** [review]

**Pull request merged**